src/libraries/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs

```diff
@@ -23,13 +21,13 @@ public void CheckSourceExistenceAndDeletion()
             finally
             {
                 EventLog.DeleteEventSource(source);
-                Helpers.Retry(() => EventLog.Delete(log));  // unlike other tests, throw if delete fails
+                if(EventLog.Exists(log))
+                    Helpers.Retry(() => EventLog.Delete(log));
```

Comment:
I know this was here before but just curious why is this being retried?

Reply:
Eventlog delete operation changes the registry and we require locks for that.  As in our tests, multiple threads are trying to achieve the same, sometimes it fails while waiting.